packages/popover/src/Popover/Popover.spec.tsx

```diff
@@ -0,0 +1,218 @@
+  });
+
+  test('does not display popover when "active" prop is not set', () => {
+    const { container } = renderPopover();
+    expect(container.innerHTML.includes('popover-test-id')).toBe(false);
+  });
+
+  test('onClick handler is called when popover contents is clicked', () => {
+    const clickSpy = jest.fn();
+    const { getByText } = renderPopover({ active: true, onClick: clickSpy });
+
+    expect(clickSpy).not.toHaveBeenCalled();
+    fireEvent.click(getByText('Popover Content'));
+    expect(clickSpy).toHaveBeenCalledTimes(1);
+  });
+
+  test('portals popover content to end of DOM by default', () => {
+    const { container, getByTestId } = renderPopover({ active: true });
+    expect(container).not.toContain(getByTestId('popover-test-id'));
+  });
+
+  test('does not portal popover content to end of DOM when "usePortal" is false', () => {
+    const { container } = renderPopover({
+      active: true,
+      usePortal: false,
+    });
+
+    expect(container.innerHTML.includes('popover-test-id')).toBe(true);
+  });
+
+  test('applies "portalClassName" to root of portal', () => {
+    const { getByTestId } = renderPopover({
+      active: true,
+      portalClassName: 'test-classname',
+    });
+
+    expect(getByTestId('popover-test-id').parentElement?.className).toBe(
+      'test-classname',
+    );
+  });
+
+  // eslint-disable-next-line jest/expect-expect
+  test('does not allow specifying "portalClassName", when "usePortal" is false', () => {
+    renderPopover({
+      active: true,
+      usePortal: false,
+      // @ts-expect-error
+      portalClassName: 'test-classname',
+    });
+  });
+
+  test('removes Popover instance on unmount', () => {
+    const { container, unmount } = renderPopover();
+    unmount();
+    expect(container.innerHTML).toBe('');
+  });
+
+  test('fires `Transition` lifecycle hooks', async () => {
+    const callbacks = {
+      onEnter: jest.fn(),
+      onEntering: jest.fn(),
+      onEntered: jest.fn(),
+      onExit: jest.fn(),
+      onExiting: jest.fn(),
+      onExited: jest.fn(),
+    };
+    const { rerenderPopover } = renderPopover({
+      ...callbacks,
+    });
+
+    // Does not call any hooks on initial render
+    for (const cb of Object.values(callbacks)) {
+      expect(cb).not.toHaveBeenCalled();
+    }
+
+    // Calls enter callbacks when active is toggled to true
+    rerenderPopover({ active: true });
+
+    expect(callbacks.onEnter).toHaveBeenCalledTimes(1);
+    expect(callbacks.onEntering).toHaveBeenCalledTimes(1);
+    await waitFor(() => expect(callbacks.onEntered).toHaveBeenCalledTimes(1));
+
+    expect(callbacks.onExit).not.toHaveBeenCalled();
+    expect(callbacks.onExiting).not.toHaveBeenCalled();
+    expect(callbacks.onExited).not.toHaveBeenCalled();
+
+    // Calls exit callbacks when active is toggled to false
+    rerenderPopover({ active: false });
+
+    expect(callbacks.onEnter).toHaveBeenCalledTimes(1);
```

Comment:
Why are these still called when toggled to false?

Reply:
Because the mock isn't reset. We expect it _only_ to have been called one time (from the previous render)

Reply:
maybe leave a comment?